File: Excel/546 Challenge.py
import numpy as np
import pandas as pd

path = "546 Pick the Odd Numbers in a Grid.xlsx"
input1 = pd.read_excel(path, header=None, usecols = "A:B", skiprows=1, nrows =2).values
input2 = pd.read_excel(path, header=None, usecols = "A:C", skiprows=4, nrows = 3).values
input3 = pd.read_excel(path, header=None, usecols = "A:C", skiprows=8, nrows = 3).values
input4 = pd.read_excel(path, header=None, usecols = "A:D", skiprows=12, nrows = 4).values
input5 = pd.read_excel(path, header=None, usecols = "A:E", skiprows=17, nrows = 5).values

test1 = pd.read_excel(path, header=None, usecols = "G", skiprows=1, nrows =1).values[0][0]
test2 = pd.read_excel(path, header=None, usecols = "G", skiprows=4, nrows =1).values[0][0]
test3 = pd.read_excel(path, header=None, usecols = "G", skiprows=8, nrows =1).values[0][0]
test4 = pd.read_excel(path, header=None, usecols = "G", skiprows=12, nrows =1).values[0][0]
test5 = pd.read_excel(path, header=None, usecols = "G", skiprows=17, nrows =1).values[0][0]


def odd_numbers(matrix):
    all = np.concatenate((matrix, np.transpose(matrix)), axis=0)
    all = np.array([int(''.join(map(str, x))) for x in all])
    return ', '.join(map(str, all[all % 2 != 0]))

print(test1 == odd_numbers(input1)) # True
print(test2 == odd_numbers(input2)) # True
print(test3 == odd_numbers(input3)) # No Value vs NaN
print(test4 == odd_numbers(input4)) # True
print(test5 == odd_numbers(input5)) # True                                                                                                  

File: Excel/500-599/546/546 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/546 Pick the Odd Numbers in a Grid.xlsx"
input1 = read_excel(path, range = "A2:B3", col_names = FALSE) %>% as.matrix()
test1  = read_excel(path, range = "G2:G2", col_names = FALSE) %>% pull()

input2 = read_excel(path, range = "A5:C7", col_names = FALSE) %>% as.matrix()
test2  = read_excel(path, range = "G5:G5", col_names = FALSE) %>% pull()

input3 = read_excel(path, range = "A9:C11", col_names = FALSE) %>% as.matrix()
test3  = read_excel(path, range = "G9:G9", col_names = FALSE) %>% pull()

input4 = read_excel(path, range = "A13:D16", col_names = FALSE) %>% as.matrix()
test4  = read_excel(path, range = "G13:G13", col_names = FALSE) %>% pull()

input5 = read_excel(path, range = "A18:E22", col_names = FALSE) %>% as.matrix()
test5  = read_excel(path, range = "G18:G18", col_names = FALSE) %>% pull()

pick_odds <- function(M) {
  all <- as.numeric(apply(rbind(M, t(M)), 1, paste0, collapse = ""))
  paste(all[all %% 2 == 1], collapse = ", ")
}

all.equal(pick_odds(input1), test1) # TRUE
all.equal(pick_odds(input2), test2) # TRUE
all.equal(pick_odds(input3), test3) # could not pull "no cells", byt result of function is also empty
all.equal(pick_odds(input4), test4) # TRUE
all.equal(pick_odds(input5), test5) # TRUE
